### PR TITLE
Add ability to change the locale.

### PR DESCRIPTION
**Pull request by mansurg**

### CODE REVIEW

File: lib/LocalizationProvider.js [hunk 1]
```diff
@@ -35,6 +35,8 @@ module.exports = LocalizationProvider;
 var util = require('util');
 
 var LOCALE_COOKIE_KEY = 'locale',
+	LOCALE_COOKIE_MAX_AGE = 3155692600, // 100 years
```
Comment (pragmadash) [inline lib/LocalizationProvider.js]: Please name with prefix `DEFAULT_`. This approach is used everywhere in Catbery.

File: lib/LocalizationProvider.js [hunk 2]
```diff
@@ -90,7 +108,23 @@ LocalizationProvider.prototype._pluralizationRulesCache = null;
...
+ */
+LocalizationProvider.prototype.changeLocale = function (locale, context) {
+	var window = context.locator.resolve('window'),
+		expireDate = new Date((new Date()).getTime() + this._cookieConfig.maxAge * 1000);
+
+	this._cookieConfig.key = this._cookieConfig.name;
+	this._cookieConfig.value = locale;
+	this._cookieConfig.expires = expireDate.toUTCString();
+	context.cookie.set(this._cookieConfig);
+	window.document.location.reload();
```
Comment (pragmadash) [inline lib/LocalizationProvider.js]: This code is executing on both server and browser environments, we should check `if (context.isBrowser)` here.

**Comment by pragmadash** [comment]
Thank you @mansurg ! 

This solution looks great, but I have a few notices in comments and please fix the [code style errors](https://travis-ci.org/catberry/catberry-l10n/jobs/54929868). Also, you can check the code style locally using the `npm test` command at the root of the repository.

**Comment by pragmadash** [comment]
Resolves #6.

File: lib/LocalizationProvider.js [hunk 3]
```diff
@@ -90,7 +109,26 @@ LocalizationProvider.prototype._pluralizationRulesCache = null;
...
+	if(context.isBrowser) {
+		var window = context.locator.resolve('window'),
+			expireDate = new Date((new Date()).getTime() +
+				this._cookieConfig.maxAge * 1000);
+
+		this._cookieConfig.key = this._cookieConfig.name;
+		this._cookieConfig.value = locale;
+		this._cookieConfig.expires = expireDate.toUTCString();
+		context.cookie.set(this._cookieConfig);
+		window.document.location.reload();
```
Comment (pragmadash) [inline lib/LocalizationProvider.js]: I meant you should put the 
```javascript
var window = context.locator.resolve('window');
window.document.location.reload();
``` 
expressions to `if(context.isBrowser)` only. 
All other `cookie` stuff will work properly on the server environment too.

All other stuff is great.

File: lib/LocalizationProvider.js [hunk 4]
```diff
@@ -90,7 +109,27 @@ LocalizationProvider.prototype._pluralizationRulesCache = null;
...
+ * @param {string} locale Locale name (i.e. en, en-us, ru etc).
+ * @param {Object} context Component context.
+ */
+LocalizationProvider.prototype.changeLocale = function (locale, context) {
+	var expireDate = new Date((new Date()).getTime() +
+			this._cookieConfig.maxAge * 1000);
+
+	this._cookieConfig.key = this._cookieConfig.name;
+	this._cookieConfig.value = locale;
+	this._cookieConfig.expires = expireDate.toUTCString();
```
Comment (pragmadash) [inline lib/LocalizationProvider.js]: One more thing I've found here: you should not use `.toUTCString()`, this parameter should be a `Date` instance.
You can found parameter types in jsDoc [here](https://github.com/catberry/catberry/blob/4.6.4/docs/index.md#cookie).

Comment (mansurg) [inline lib/LocalizationProvider.js]: I will try to be considerate.

**Comment by pragmadash** [comment]
:+1:

**Pull request merged**